src/query/catalog/src/table_context.rs

```diff
@@ -203,12 +203,13 @@ pub trait TableContext: Send + Sync {
     fn get_current_session_id(&self) -> String {
         unimplemented!()
     }
-    async fn get_all_effective_roles(&self) -> Result<Vec<RoleInfo>>;
+    async fn get_all_effective_roles(&self, create_object: bool) -> Result<Vec<RoleInfo>>;
 
     async fn validate_privilege(
         &self,
         object: &GrantObject,
         privilege: UserPrivilegeType,
+        create_object: bool,
```

Comment:
is this parameter can be removed? can we directly use the `privilege` parameter to check it's a CREATE XXX operation or not?

Reply:
Now create udf/stage/task need super privilege. So I think we can use parameter tmp. Until we support Create xxObject privilege then refactor this code.

Reply:
And this check, only need to check ownership object's created.

Reply:
can also consider rename the `create_object` here as `check_current_role_only`, to describe what this function does instead of emphasis it's a special case.